### PR TITLE
input-group: add validation section

### LINKED ISSUE #3146: Update igx-input-group topic with validation examples
The topic should include validation examples with `igx-input-group` components in a form. It should include examples and samples with both template-driven and reactive forms, as well as examples of custom form validation (reactive forms). 

Here's the form validation topic in angular for reference: https://angular.io/guide/form-validation

### PR DESCRIPTION
**Pull request by RivaIvanova**
Closes #3146 

### Checklist:
 
 - [ ] check topic's TOC/menu and paragraph headings
 - [ ] link to other topics using `../relative/path.md`
 - [ ] at the References section at the end of the topic add links to topics, samples, etc
 - [ ] reference API documentation instead of adding a section with API

<br />

 - [x] use valid component names - [Data] Grid, `IgxSelectComponent`, `<igx-combo>`
 - [x] use spell checker tool (VS Code, Grammarly, Microsoft Editor)
 - [x] add inline `code blocks` for the names of classes / tags / properties
 - [x] add language descriptor for the ```code blocks```
 - [ ] check broken links (use browser add-on)
 - [x] check if sample is working and fully visible in the topic
 - [x] check if sample is working and fully visible in the StackBlitz
 - [ ] check if code blocks match the code in StackBlitz demo

<br />

 - [x] follow SEO guidelines to fill topic's metadata ([SEO guidelines](https://infragisticsinc297.sharepoint.com/:w:/g/Groups/marketing/EWz9InT4FDlErHCumxsKGY4Bd8H03yhRWxDFk47luRz-_Q?e=S5wWcx))

<br />

 - [ ] do not resolve requested changes (leave that to the reviewer)
 - [ ] add `pending-localization` label when the review of the PR is done
 - [ ] add a member from the localization team to translate it

### CODE REVIEW

File: en/components/input-group.md [hunk 1]
```diff
@@ -131,6 +131,142 @@ The Ignite UI for Angular Input Group component can be used inside strictly type
            iframe-src="{environment:demosBaseUrl}/data-entries/typed-form" >
 </code-view>
 
+## Validation
+The following samples demonstrate how input validation could be achieved when using template-driven or reactive forms.
```
Comment (kdinev) [inline en/components/input-group.md:135]:
```suggestion
The following samples demonstrate how to configure input validation when using [template-driven](https://angular.io/guide/forms) or [reactive forms](https://angular.io/guide/reactive-forms).
```

File: en/components/input-group.md [hunk 2]
```diff
@@ -131,6 +131,142 @@ The Ignite UI for Angular Input Group component can be used inside strictly type
...
+
+### Template-Driven Forms
+Template-driven form validation is achieved by adding validation attributes, i.e., `required`, `minLength`, etc., to the `input` element.
+
+```html
+<igx-input-group>
```
Comment (kdinev) [inline en/components/input-group.md:141]: Include the wrapping form element in each code snippet, together with the submit button.

File: en/components/input-group.md [hunk 3]
```diff
@@ -131,6 +131,142 @@ The Ignite UI for Angular Input Group component can be used inside strictly type
...
+
+<igx-input-group>
+    <label igxLabel for="password">Password</label>
+    <input igxInput name="password" type="password" required minlength="8" />
+</igx-input-group>
+```
+
+The `required` attribute adds an asterisk next to the label, indicating that this field must be completed, however, when having additional validation like `email` and `minLength`, the user could be notified via the [`igx-hint`]({environment:angularApiUrl}/classes/igxhintdirective.html) directive.
+
+The following example uses two-way data binding and demonstrates how to inspect the control's state by exporting the `ngModel` to a local variable.
```
Comment (kdinev) [inline en/components/input-group.md:159]: Also include an example on how to inspect general form validity through the ngForm. In general the check there would be perform in the submit button to make it enabled/disabled.

File: en/components/input-group.md [hunk 4]
```diff
@@ -131,6 +131,142 @@ The Ignite UI for Angular Input Group component can be used inside strictly type
...
+```ts
+public registrationForm = new FormGroup<User>({
+    username: new FormControl<string>('', { nonNullable: true, validators: [Validators.required] }),
+    email: new FormControl<string>('', { nonNullable: true, validators: [Validators.required, Validators.email] }),
+    password: new FormControl<string>('', { nonNullable: true, validators: [Validators.required, Validators.minLength(8)] })
+});
+```
+
+```html
+<form [formGroup]="registrationForm">
```
Comment (kdinev) [inline en/components/input-group.md:194]: Again, include a submit button.

File: en/components/input-group.md [hunk 5]
```diff
@@ -131,6 +131,142 @@ The Ignite UI for Angular Input Group component can be used inside strictly type
...
+
+The result from the above configurations could be seen in the below sample.
+
+<code-view style="height:480px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/data-entries/reactive-form-validation" >
+</code-view>
+
+### Custom Validators
+Some input fields may require custom validation and this could be achieved via custom validators. When the value is valid, the validator will return `null` and when the value is invalid, it will generate a set of errors that could be used to display a specific error message on the screen.
```
Comment (kdinev) [inline en/components/input-group.md:247]: I have left a review comment about the custom validation sample in the samples repo. There, you would be able to create an example of how one from control drives the validity of a different form control, or of the entire group.

**Review by kdinev — CHANGES_REQUESTED** [review]

**Review by jackofdiamond5 — APPROVED** [review]

**Pull request merged**